torch/csrc/jit/python/pybind_utils.h

```diff
@@ -224,6 +224,7 @@ struct VISIBILITY_HIDDEN PythonFutureWrapper
   // vector that has exactly one tensor.
   static std::vector<std::reference_wrapper<const at::DataPtr>> dataPtrExtractor(
       const at::IValue& value) {
+    TORCH_INTERNAL_ASSERT(PyGILState_Check() == 0);
```

Comment:
I also added this because I was seeing some deadlocks (before I fixed the segfault) which seemed unrelated, and my suspicion was that it came from a reentrant GIL acquisition, possibly due to the DataPtr extractor now being used in more places. However the CI at the moment doesn't seem to trigger neither the deadlock nor this check.